assets/js/script.js

```diff
@@ -0,0 +1,162 @@
+let darkMode = false;
+
+
+window.onload = () => {
+    fetch('https://api.github.com/users/octocat')
+    .then(response => response.json())
+    .then(data => {
+        showUserData(data)})
+    .catch(error => {
+        throw error;})
+}
+
+button.addEventListener('click', (e)=> {
+    if (input.value != "") {
+        userData(gitUrl+input.value)
+    } 
+})
+
+input.addEventListener('keydown', (e) => {
+    if (e.key == 'Enter') {
+        if (input.value != "") {
+            userData(gitUrl+input.value)
+        } 
+    }
+});
+
+input.addEventListener ('input', (e) => {
+    errorMessage.style.display= 'none';
+})
+
+const userData = (inputUrl) => {
+    fetch(inputUrl)
+    .then(response => response.json())
+    .then(data => {
+        showUserData(data)})
+    .catch(error => {
+        throw error;})
+}
+
+const showUserData = (data) => {
+    //console.log(data);
+    errorMessage.style.display = 'none';
+    if(data.message != 'Not Found'){
+        profileImg.src = data.avatar_url;
+        profileName.innerHTML = data.name;
+        let formatedDate = data.created_at.split('T').shift().split('-');
+        //console.log(formatedDate);
+        joinDate.innerText = 'Joined '+ formatedDate[2] + ' ' +(months[formatedDate[1]-1]) + ' ' + formatedDate[0];
+        userName.innerText = '@'+data.login;
+        userName.href = data.html_url;
+        bio.innerText = (data.bio == null) ? "This profile has no bio" : data.bio;
+        repos.innerText = data.public_repos;
+        followers.innerText = data.followers;
+        following.innerText = data.following;
+        const checkIfDataIsNull = (receivedData, element) => {
+            if (receivedData == "" || receivedData == null){
+                element.style.opacity = 0.5;
+                element.previousElementSibling.style.opacity = 0.5;
+                return "Not available"
+            }else {
+                element.style.opacity = 1;
+                element.previousElementSibling.style.opacity = 1;
+                return receivedData;
+            }
+        }
+        userLocation.innerText = checkIfDataIsNull(data.location, userLocation);
+        twitter.innerText = checkIfDataIsNull(data.twitter_username, twitter);
+        if (twitter.innerHTML != "Not available" || twitter.href == 'undefined') {
+            twitter.href ='https://twitter.com/'+data.twitter_username;
+        } else {
+            twitter.removeAttribute('href');} 
+        blog.innerText = checkIfDataIsNull(data.blog, blog);
+        if (blog.innerHTML != "Not available" || blog.href == 'undefined') {
+            blog.href = data.blog;
+        } else {
+            blog.removeAttribute('href');
+        }
+        company.innerText = checkIfDataIsNull(data.company, company);
+
+    }else {
+        errorMessage.style.display = 'block';
+    }
+}
+
+lightDarkSwitch.addEventListener('click', (e)=> {
+    if(darkMode == false) {
+        trans();
```

Comment:
Great! It's good that you're calling a function like this, so we can use it multiple times without duplicating the code!